updated worker api see issue 255

RyanMTB opened the pull request:
This is a fix to: https://github.com/nvie/rq/issues/255

Now all of these will work.

```python
w = Worker('high')  

q = Queue(name = 'high')
w = Worker(q)

queue_names = ['high', 'normal', 'default']
w = Worker(queue_names)
w = Worker((map(Queue, queue_names))
w = Worker((list(map(Queue, queue_names)))

w = Worker([queue_names, q])
```

This is my first ever attempt at contributing to an open source project. All feedback is greatly appreciated.

selwin commented on rq/worker.py:
```diff
@@ -160,11 +161,22 @@ def __init__(self, queues, name=None,
+
+    def process_queue_args(self, queue_args):
+        """ allow for a variety arguements to queues """
+        if isinstance(queue_args, six.text_type):
```
Can you please change this line to use `rq.compat.text_type` so we don't need need to depend on `six`?

selwin commented:
Thanks for the PR @RyanMTB ! I'll merge this in when we lose the `six` dependency.

selwin commented on rq/worker.py:
```diff
@@ -160,11 +161,22 @@ def __init__(self, queues, name=None,
+            queue_args = [[queue_arg] if isinstance(queue_arg, six.text_type) 
+                    else list(queue_arg) for queue_arg in queue_args]
+            queue_args = itertools.chain.from_iterable(queue_args)
+            return  [self.queue_class(name = queue_arg) if isinstance(queue_arg,six.text_type) 
```
Nitpicks: could you please remove the spaces between the `=` in `self.queue_class(name = queue_arg)`? Also, put a space after comma in `isinstance(queue_arg,six.text_type)`

RyanMTB commented:
@selwin Fixed.  Thank you for the feedback and comments.

My pull request somehow now contains my upstream/master merge commit.  Let me know if that is a problem.  I am not sure.

selwin commented on rq/worker.py:
```diff
@@ -160,11 +160,22 @@ def __init__(self, queues, name=None,
         for queue in self.queues:
             if not isinstance(queue, self.queue_class):
-                raise NoQueueError('Give each worker at least one Queue.')
+                raise NoQueueError('Give each worker at least one Queue.\
```
I know this error message has been here from the very start. Doesn't `Give each worker at least one Queue` sound a bit strange? I think we should just change the error message to `{} is not a Queue`.

selwin commented on rq/worker.py:
```diff
@@ -160,11 +160,21 @@ def __init__(self, queues, name=None,
+        if isinstance(queue_args, text_type):
+            return self.queue_class(name = queue_args)
+        else:
+            queue_args = [queue_args] if isinstance(queue_args, text_type) else queue_args
```
This line of code will never be used since we already checked for `text_text` instance in [Line 169](https://github.com/nvie/rq/pull/531/files#diff-1aac421e8c53e895ee40f3b7dc63144fR169)

RyanMTB replied:
I think we still need it. queue_args could be a list and there is still a chance for the user to pass in something that is not a queue.

As the code is currently written it will flatten iterables that are nested one deep. (maybe this is unnecessary?)

```Python
Worker([['foo']]) #this will work
Worker([[['foo']]]) #this will not work
```
I would be happy to write more tests if that would clear things up.

RyanMTB commented:
I cleaned up my code so it will no longer flatten nested lists(unnecessary).  I also added a test to show that the following still works:

```Python
w = Worker(list(map(Queue, ['high', 'low']))
```

selwin commented:
It looks much simpler now, very nice! Thanks @RyanMTB!
